### PR TITLE
Show error when trying to continue after successful command

### LINKED ISSUE #364: `--continue` results in ugly error message if there is no continue script
Actual behavior:
```
~/D/b/git-town [kg-town-error-code] > git ship --continue
/Users/kevin/Dropbox/bin/git-town/src/helpers/script_helpers.sh: line 15: continue_: command not found
```

Expected behavior:
```
~/D/b/git-town [kg-town-error-code] > git ship --continue
Error: The last command finished successfully, and cannot be continued
```

### PR DESCRIPTION
**Pull request by ricmatsui**
Closes #364

### CODE REVIEW

**Comment by charlierudolph** [comment]
Can you please add a feature test for this. Since this shared behavior between most of the commands I think it would be good to but this under something like
```
features/shared/continuing_after_success.feature
```
Should probably add one for abort as well. Though that can be a separate PR

**Comment by ricmatsui** [comment]
No problem. I have added a simple test for after using git-ship. Let me know if we should have more for the other commands.

File: features/shared/continuing_after_success.feature [hunk 1]
```diff
@@ -0,0 +1,18 @@
+Feature: Show clear error if trying to continue after executing a successful command
+
+  As a developer accidentally trying to continue a command after it completed successfully
+  I should see a clear error that the command cannot be continued
```
Comment (kevgo) [inline features/shared/continuing_after_success.feature]: I should see a friendly and descriptive message that the command can not be continued

File: features/shared/continuing_after_success.feature [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  So that I don't experience any cryptic errors.
```
Comment (kevgo) [inline features/shared/continuing_after_success.feature]: So that I have fun learning how to use Git Town correctly.

File: features/shared/continuing_after_success.feature [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  Background:
+    Given I have a feature branch named "feature"
+    And the following commits exist in my repository
+      | BRANCH  | LOCATION         |
+      | feature | local and remote |
+    And I run `git ship -m 'feature done'`
+
+
+  Scenario: continuing after successful ship
+    Then I run `git ship --continue`
```
Comment (kevgo) [inline features/shared/continuing_after_success.feature]: *When* I run ...

File: features/shared/continuing_after_success.feature [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    And I get the error "The last command finished successfully and cannot be continued"
```
Comment (kevgo) [inline features/shared/continuing_after_success.feature]: *Then* I get the error ...

**Comment by kevgo** [comment]
I was thinking about adding a feature spec as well. Great solution! Let's add one spec per command, just to make clear that it applies to each one.

File: features/shared/continuing_after_success.feature (hunk 2 above)
Comment (allewun) [inline features/shared/continuing_after_success.feature]: I think @ricmatsui's original sentence makes more sense. Maybe in addition to to *cryptic* -- *irrelevant*?

**Comment by ricmatsui** [comment]
All comments addressed besides the feature preamble reason.

File: features/shared/continuing_after_success.feature (hunk 2 above)
Comment (kevgo) [inline features/shared/continuing_after_success.feature]: Okay, I'm convinced. Fine to leave it as is.

File: features/shared/continuing_after_success.feature [hunk 5]
```diff
@@ -0,0 +1,73 @@
+Feature: Show clear error if trying to continue after executing a successful command
```
Comment (kevgo) [inline features/shared/continuing_after_success.feature]: Cool idea to create a "shared" feature folder. This prevents us from having to repeat such concerns that are orthogonal to the individual commands unnecessarily. Let's keep our eyes open for opportunities to dry up some other orthogonal features into this folder!

**Comment by kevgo** [comment]
:ship: This turned out really nice!

**Comment by allewun** [comment]
Yep, this is nice! :ship:
